Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Types/Booking.hs

```diff
@@ -76,3 +78,103 @@ data Booking = Booking
 
 data BookingType = SpecialZoneBooking | NormalBooking
   deriving (Show, Eq, Ord, Read, Generic, ToJSON, FromJSON, ToSchema)
+
+data BookingTable = BookingTable
```

Comment:
What is this?

Reply:
To handle -> we removed from and to location Id from Booking tables, but it has to be a part of Domain Types. We cant do toTType with missing values in table.
to handle this we created a separate domain/types BookingTable which will do fromTType then convert this back to Booking type

Reply:
> To handle -> we removed from and to location Id from Booking tables, but it has to be a part of Domain Types. We cant do toTType with missing values in table. to handle this we created a separate domain/types BookingTable which will do fromTType then convert this back to Booking type

All things you are doing in `bookingTableToBookingConverter` can be done in builder layer, it is designed to work as converter. I don't understand what exactly you can't do in here so you had to create this converter thing

Reply:
> what is the builder layer, can you elaborate?

It is [this](https://github.com/nammayatri/nammayatri/pull/1205/files#diff-c079368122def32cf9ede457599e20d2f4448471dc234d0c0b131d4ecad59225L49-L59) part you just removed

Reply:
actually the requirement was also to remove all joins.. thats why we didnt join the location tables with booking/searchRequest

Reply:
> actually the requirement was also to remove all joins.. thats why we didnt join the location tables with booking/searchRequest

You don't need joins to work with this.

Reply:
```
buildFullBooking ::
  Transactionable m =>
  (BookingT, BookingLocationT, Maybe BookingLocationT, Maybe TripTermsT, Maybe RentalSlabT) ->
  DTypeBuilder m (Maybe (SolidType FullBookingT))
```
You mean here we should take BookingTable and return Booking??

Reply:
> ```
> buildFullBooking ::
>   Transactionable m =>
>   (BookingT, BookingLocationT, Maybe BookingLocationT, Maybe TripTermsT, Maybe RentalSlabT) ->
>   DTypeBuilder m (Maybe (SolidType FullBookingT))
> ```
> 
> You mean here we should take BookingTable and return Booking??

No i mean you should do you selects from new location table in this func. And `FullBookingT` will be 
`type FullBookingT = (BookingT, SLoc.LocationT, Maybe STripTerms.TripTermsT, BookingDetailsT)`
i guess

Reply:
but why do we need fullbookingT as a part of the task was to remove all joins as well